paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -330,7 +336,7 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
         "dilation in convolution is not implemented yet");
 
     const T* input_data = input->data<T>();
-    const T* filter_data = filter->data<T>();
+    const float* filter_data = filter->data<float>();
```

Comment:
Is there any particular reason why this has been changed?

Reply:
yes, we hard code the type of filter data for INT8, because filter data type should always be float, while INT8 will use u8 or s8 as "T".

Reply:
@panyx0718 Do I understand correctly that your idea would be to have additional template parameter in ConvMKLDNNKernel class for type of filter data?

Reply:
Right, that seems to be the standard way. There might be a few framework changes needed to support multiple template types and select the right kernel at runtime. @wanghaoshuang can help with that.

Reply:
We may adapt the change later once framework provides the new template; it does not block this PR.

Reply:
https://github.com/PaddlePaddle/Paddle/pull/14742

I think the kernel to have 2 template fields, one for Input, one for Filter.
You need to REGISTER several different kernels and give them special name and value
REGISTER(kernel<float, int8>, FLOAT_INT8, 1)
REGISTER(kernel<float, float>, FLOAT_FLOAT, 2)
REGISTER(kernel<int8, int8>, INT8_INT8, 3)

In GetExpectedKernelType, you init the OpKernelType(....,1/2/3) based on ctx.Input type